packages/react/ds-app-launchpad/src/ui/FileTree/Provider.tsx

```diff
@@ -0,0 +1,127 @@
+import type React from "react";
+import { useCallback, useEffect, useRef } from "react";
+import Context from "./Context.js";
+import "./styles.css";
+import type { ManagedContextOptions, ProviderOptions } from "./types.js";
+
+const componentCssClassName = "ds file-tree";
+
+const Provider = ({
+  id,
+  className,
+  style,
+  children,
+  ...contextOptions
+}: ProviderOptions): React.ReactElement => {
+  // const focusedNodeRef = useRef<HTMLLIElement>(null);
+  const fileTreeRef = useRef<HTMLDivElement>(null);
+
+  const handleFocusNextNode: ManagedContextOptions["focusNextNode"] =
+    useCallback((direction) => {
+      if (!fileTreeRef.current) return;
+      const currentNode = fileTreeRef.current.querySelector<HTMLElement>(
+        "[data-path-hash]:focus",
```

Comment:
`data-path-hash` literal is repeated several times, I'd consider extracting it to a constant